test/Storage.js

```diff
@@ -278,5 +278,128 @@ describe('Storage', () => {
           }
       });
     });
-  })
+  });
+
+  describe('listDataBySensor', () => {
+    let storage, query;
+    const deviceId = '0643ca1462f94b79';
+    const sensorId = 1;
+
+    describe('when deviceId is invalid', () => {
+      beforeEach(() => {
+        storage = new Storage(settings);
+      });
+
+      it('should throw an exception', async () => {
+        expect(storage.listDataBySensor('4797@$fa', sensorId))
+          .to.eventually.be.rejected;
+      });
+    });
+
+    describe('with empty query', () => {
+      beforeEach(() => {
+        storage = new Storage(settings);
+        nock(url)
+          .get(`/data/${deviceId}/sensor/${sensorId}`)
+          .reply(200, dataList);
+      });
+
+      it('should returns an array', async () => {
+        const response = await storage.listDataBySensor(deviceId, sensorId);
+        expect(response).to.be.an('array');
+      });
+
+      it('should returns an array of data objects', async () => {
+        const response = await storage.listDataBySensor(deviceId, sensorId);
+        expect(response).to.deep.equal(dataList);
+      });
+    });
+
+    describe('with date interval query', () => {
+      let dataBetweenTimeInterval;
+
+      beforeEach(() => {
+        query = { startDate: '2019-03-18T14:42:03.192Z', finishDate: '2019-03-18T16:17:45.192Z'};
+        dataBetweenTimeInterval = dataList.slice(1, 4);
+        storage = new Storage(settings);
+        nock(url)
+          .get(`/data/${deviceId}/sensor/${sensorId}`)
+          .query(query)
+          .reply(200, dataBetweenTimeInterval);
+        });
+
+      it('should returns an array', async () => {
+        const response = await storage.listDataBySensor(deviceId, sensorId, query);
+        expect(response).to.be.an('array');
+      });
+
+      it('should returns an array of data objects between time interval', async () => {
+        const response = await storage.listDataBySensor(deviceId, sensorId, query);
+        expect(response).to.deep.equal(dataBetweenTimeInterval);
+      });
+    });
+
+    describe('with take and skip query', () => {
```

Comment:
See discussions about tests in previous commits.

Reply:
Waiting for response.

Reply:
Done.